Source/WebCore/rendering/RenderBlock.cpp

```diff
@@ -2492,7 +2492,7 @@ LayoutUnit RenderBlock::minLineHeightForReplacedRenderer(bool isFirstLine, Layou
 
 std::optional<LayoutUnit> RenderBlock::firstLineBaseline() const
 {
-    if (shouldApplyLayoutContainment())
+    if (shouldApplyLayoutContainment() || style().containerType() != ContainerType::Normal)
```

Comment:
I would run that through @alanbaradlay or @bfgeek . It definitely doesn't seem right to branch on containerType here.

Without this change, some table/table-cell subtests on http://wpt.live/css/css-conditional/container-queries/nested-query-containers.html fail. 

Does baseline computation change with independent formatting contexts? or is it something else? I wonder how Blink gets away with not having to do a similar change.

Reply:
Yeah this is incorrect with this you'll likely fail something like:

```
<div style="display: flex; align-items: baseline;">
<div>baseline</div> <div style="container-type: inline-size; width: 100px;">aligned</div>
</div>
```

If its table cell + baseline specific I would start looking at hysterisis with intrinsic-padding:
https://github.com/WebKit/WebKit/blob/e3f070792f2ec7eff5fa3dabc4bde4732b70260d/Source/WebCore/rendering/RenderTableCell.cpp#L231

Reply:
Thanks!

Reply:
This seems to boil down to this very bug where cell sizing extract baseline information from the "contain: size" child and sizes the cell accordingly. This fails on shipping Safari (and Firefox too), while Chrome passes.

```
<style>
.child {
  contain: size;
}
</style>
<table id=table1>
  <td style="vertical-align: baseline">
    <div class=child>&nbsp</div>
  </td>
</table>
<table id=table2>
  <td style="vertical-align: middle">
    <div class=child>&nbsp</div>
  </td>
</table>
<pre id=result>PASS if true: </pre>

<script>result.innerText += table1.offsetHeight == table2.offsetHeight</script>
```

Reply:
If I had to guess - i'd start by looking at - https://github.com/WebKit/WebKit/blob/964f4b1be9b5c53988d71c1029b2dc27c198cf16/Source/WebCore/rendering/RenderTableSection.cpp#L317

e.g. logic in that area is assuming that `baselines > 0` which isn't strictly the case.
For that section its doing a `height - baseline == 0 - 12`

Reply:
(e.g. you get the same result in the testcase if `contain: size` is replaced with `height: 0`)

Reply:
and this is not specific to tables. Flex fails the same way when baseline aligned, unless I am holding it wrong and baseline alignment assumes enclosing behavior (well it does not).

```
<style>
.wrapper {
  display: flex;
  outline: 1px solid red;
}
.inner {
  height: 0px;
}
</style>
<div class="wrapper" style="align-items: center">
  <div class="inner">&nbsp</div>
</div>
<div class="wrapper" style="align-items: baseline">
  <div class="inner">&nbsp</div>
</div>

```

Reply:
Yeah Blink/WebKit are incorrect there, Firefox is correct.

For us - https://chromium-review.googlesource.com/c/chromium/src/+/5901280 fixes it (renders the 2nd flexbox as zero).

Reply:
> For that section its doing a `height - baseline == 0 - 12`

Was meant to say `height - baseline == 0 - (-12)` :)